Core/AppRuntime/Source/WorkQueue.cpp

```diff
@@ -42,6 +43,15 @@ namespace Babylon
         while (!m_cancelSource.cancelled())
         {
             m_dispatcher.blocking_tick(m_cancelSource);
+
+            // Call the unhandled exception handler if an exception is thrown in one of the tasks.
+            m_task = m_task.then(arcana::inline_scheduler, arcana::cancellation::none(), [this](const arcana::expected<void, std::exception_ptr>& result) {
+                if (result.has_error())
+                {
+                    m_unhandledExceptionHandler(result.error());
```

Comment:
It looks like providing this as an argument to the `WorkQueue` constructor is optional? Do we need to check it before calling it?

Reply:
The default version points to a default function. `m_unhandledExceptionHandler` is always set to something.